spec/index.emu

```diff
@@ -107,6 +107,27 @@ location: https://rawgit.com/tc39/proposal-realms/master/index.html
           </emu-note>
       </emu-clause>
 
+      <emu-clause id="sec-realm.prototype.import">
+          <h1>Realm.prototype.import ( _specifier_ )</h1>
+
+          The following steps are taken:
+
+          <emu-alg>
+          1. Let _O_ be *this* value.
+          1. Perform ? RequireInternalSlot(_O_, [[Realm]]).
+          1. Let _referencingScriptOrModule_ be *null*.
```

Comment:
@erights @littledan @ljharb this is an interesting question to be answered. What is the referencing scriptOrModule when calling `r.import()`? should be null? should be the code that creates the Realm object? if so, what are the implications?

Reply:
Should it be an optional parameter passed to `Realm.prototype.import`? otherwise it'd have to be "null", because this isn't syntactic, it's API.

Reply:
The `import` method called here is a first class function, not a special form. When it calls the underlying operation, it should pass null as the referrer. The first class `import` method should not take an optional referrer parameter, nor should it provide a referrer based on any relationship to the creator of the Realm or Compartment. Just null.

Reply:
Notes from SES meeting (yesterday) about this question:

* @erights convinced me that a second argument is problematic. the convincing factor was that if evaluation is available, then `myRealm.globalThis.eval('import("./something")')` should work, and at that point, you will have no way to provide a value to the referencing `scriptOrModule`.
* we need to decide whether the value should be `*null*` or `*undefined*`.
* in the meeting, we reached consensus about the behavior here, the realm object will ask the host for the resolution, by passing null, and it is the host's responsibility to decide what value to use for the referencing `scriptOrModule`.
* eventually, once we have a way to control the host (via compartment/evaluator), you will be able to control the behavior of such resolution.

@erights did I miss anything here?

Reply:
Regarding the Realm proposal, yes, I think that's everything relevant.

For reference, the meeting is at https://www.youtube.com/watch?v=hqKGKDw-FCY&list=PLzDw4TTug5O1jzKodRDp3qec8zl88oxGd which includes further considerations relevant to the SES / Compartment proposal.

Reply:
> @erights convinced me that a second argument is problematic. the convincing factor was that if evaluation is available, then myRealm.globalThis.eval('import("./something")') should work, and at that point, you will have no way to provide a value to the referencing scriptOrModule.

So what did you think of my idea to provide a "base" string to hosts?

> we need to decide whether the value should be *null* or *undefined*.

I think the choice of *null* vs *undefined* is just editorial--this is only visible to hosts. I'm fine with either.

> in the meeting, we reached consensus about the behavior here, the realm object will ask the host for the resolution, by passing null, and it is the host's responsibility to decide what value to use for the referencing scriptOrModule.

I think we should think through what to do for some environments before Stage 3 (#225). For HTML, for example, do you think this should be relative to where the whole page is, or should paths be forced to be absolute? And what should this be in Node.js? (I think the experience with ES6 modules showed that putting these decisions off has cost.)

> eventually, once we have a way to control the host (via compartment/evaluator), you will be able to control the behavior of such resolution.

Would this have a way to tell the host the "base URL" or equivalent, for environments where this is meaningful? Are there proposals that I could review on this?

> For reference, the meeting is at ...

FWIW, if there are written notes, these are generally easier for me to follow than a video.

Reply:
@littledan yes, we discussed the resolution in browsers as well, sorry I didn't add that part to the relevant notes above. The reasoning was that doing `r.import('...')` works just like doing `<script type="module" src="...">` in browsers. In other words, the module that creates the realm does not have a saying on the resolution or base referral definition. These two features will be available as part of the compartment/evaluator where you control every aspect of the host for any realm object underneath. In other words, "relative to where the whole page" if the specifier is relative. If it is absolute, then the base has no saying in the resolution.

As for the notes, no, we don't have written notes for SES meetings as far as I can tell, only the video recorded by @erights